src/Servers/HttpSys/test/FunctionalTests/Listener/ResponseHeaderTests.cs

```diff
@@ -207,20 +208,85 @@ public async Task ResponseHeaders_11HeadRequestStatusCodeWithoutBody_NoContentLe
     }
 
     [ConditionalFact]
-    public async Task ResponseHeaders_HTTP10KeepAliveRequest_Gets11Close()
+    public async Task ResponseHeaders_HTTP10KeepAliveRequest_KeepAliveHeader_RespectsSwitch()
+    {
+        AppContext.SetSwitch("Microsoft.AspNetCore.Server.HttpSys.RespectHttp10KeepAlive", true);
```

Comment:
I remember there being some complexity here around interactions between different tests.  That is, if tests are running in parallel, I don't think we want this changing states.  I'm pretty sure Chris suggested I create an internal property populated from the context switch but overridable by tests.

Reply:
See `KestrelServerOptions.FinOnError`.

Reply:
for `EnableKernelResponseBuffering`, I made use of an `internal` property on `HttpSysOptions`, that is initialized in the .ctor via the app-context switch - that way, relevant tests can set it explicitly via the `internal` property rather than the app-context switch

but yes @amcasey  you're right that changing app-context directly in tests makes for a race condition in parallel tests